src/sgraph.cxx

```diff
@@ -23,7 +23,7 @@ namespace ifc {
         // equal to that entry's position in the table, after appropriate isomorphic conversion.
         // When the predicate holds at compile time, that implies runtime look up by key
         // can be implemented by a simple constant-time indexing instead of an elaborate search.
-        template<typename T, int N>
+        template<typename T, std::size_t N>
```

Comment:
Let's use `auto` here instead of `std::size_t` to avoid implicit conversion on the explicit template argument.  Several instances below.

Reply:
While that compiles, it makes VS IntelliSense freak out:

```
no instance of function template matches the argument list
argument types are: (const ifc::<unnamed>::SortNameMapEntry<ifc::DeclSort> [32])
```

Reply:
> While that compiles, it makes VS IntelliSense freak out:
> 
> ```
> no instance of function template matches the argument list
> argument types are: (const ifc::<unnamed>::SortNameMapEntry<ifc::DeclSort> [32])
> ```

That is fine.  That is VS bug  in C++20 that the VS people should know about  :-)